pydantic/json_schema.py

```diff
@@ -21,13 +21,15 @@
     cast,
 )
 
-from pydantic_core import CoreSchema, CoreSchemaType, core_schema
+import pydantic_core
 from typing_extensions import Literal, TypedDict
 
 from ._internal import _core_metadata, _core_utils, _typing_extra, _utils
 from .errors import PydanticInvalidForJsonSchema, PydanticUserError
 
 if TYPE_CHECKING:
+    from pydantic_core import CoreSchema, CoreSchemaType, core_schema
```

Comment:
Is there a good reason to have both this and the `import pydantic_core` above? Is it to prevent adding things to the namespace that could be imported?

Reply:
yes. I'm not entirely clear how important this cleanliness is, but it's not very difficult.